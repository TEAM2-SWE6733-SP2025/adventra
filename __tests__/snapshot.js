import { render } from "@testing-library/react";
import Page from "../src/app/page";
import { SessionProvider } from "next-auth/react";

it("renders homepage unchanged", () => {
  const mockSession = {
    user: {
      name: "Test User",
      email: "test@example.com",
      image: "https://example.com/avatar.png",
    },
    expires: "2025-12-31T23:59:59.999Z",
  };

  const { container } = render(
    <SessionProvider session={mockSession}>
      <Page />
    </SessionProvider>,
  );

  expect(container).toMatchSnapshot();
});
