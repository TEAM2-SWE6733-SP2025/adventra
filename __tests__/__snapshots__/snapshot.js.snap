// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <main
    class="relative w-full h-screen overflow-hidden flex flex-col bg-black"
    style="background-image: url(/scuba.jpg); background-size: cover; background-position: center;"
  >
    <header
      class="fixed top-0 left-0 w-full z-50 transition-transform duration-300 translate-y-0"
    >
      <nav
        class="bg-white dark:bg-black border-b border-gray-200 dark:border-gray-700 w-full"
      >
        <div
          class="max-w-7xl mx-auto px-6 sm:px-8 lg:px-10"
        >
          <div
            class="flex justify-between items-center h-16"
          >
            <div
              class="flex-shrink-0 flex items-center space-x-2"
            >
              <a
                class="lg:px-5"
                href="/preferences"
              >
                <svg
                  class="w-5 h-5 text-yellow-500 hover:text-yellow-700"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M495.9 166.6c3.2 8.7 .5 18.4-6.4 24.6l-43.3 39.4c1.1 8.3 1.7 16.8 1.7 25.4s-.6 17.1-1.7 25.4l43.3 39.4c6.9 6.2 9.6 15.9 6.4 24.6c-4.4 11.9-9.7 23.3-15.8 34.3l-4.7 8.1c-6.6 11-14 21.4-22.1 31.2c-5.9 7.2-15.7 9.6-24.5 6.8l-55.7-17.7c-13.4 10.3-28.2 18.9-44 25.4l-12.5 57.1c-2 9.1-9 16.3-18.2 17.8c-13.8 2.3-28 3.5-42.5 3.5s-28.7-1.2-42.5-3.5c-9.2-1.5-16.2-8.7-18.2-17.8l-12.5-57.1c-15.8-6.5-30.6-15.1-44-25.4L83.1 425.9c-8.8 2.8-18.6 .3-24.5-6.8c-8.1-9.8-15.5-20.2-22.1-31.2l-4.7-8.1c-6.1-11-11.4-22.4-15.8-34.3c-3.2-8.7-.5-18.4 6.4-24.6l43.3-39.4C64.6 273.1 64 264.6 64 256s.6-17.1 1.7-25.4L22.4 191.2c-6.9-6.2-9.6-15.9-6.4-24.6c4.4-11.9 9.7-23.3 15.8-34.3l4.7-8.1c6.6-11 14-21.4 22.1-31.2c5.9-7.2 15.7-9.6 24.5-6.8l55.7 17.7c13.4-10.3 28.2-18.9 44-25.4l12.5-57.1c2-9.1 9-16.3 18.2-17.8C227.3 1.2 241.5 0 256 0s28.7 1.2 42.5 3.5c9.2 1.5 16.2 8.7 18.2 17.8l12.5 57.1c15.8 6.5 30.6 15.1 44 25.4l55.7-17.7c8.8-2.8 18.6-.3 24.5 6.8c8.1 9.8 15.5 20.2 22.1 31.2l4.7 8.1c6.1 11 11.4 22.4 15.8 34.3zM256 336a80 80 0 1 0 0-160 80 80 0 1 0 0 160z"
                  />
                </svg>
              </a>
              <a
                class="text-2xl font-bold"
                href="/"
              >
                Adventra
              </a>
            </div>
            <button
              class="md:hidden text-gray-500 dark:text-gray-300 hover:text-gray-700 dark:hover:text-gray-100 cursor-pointer"
            >
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 6h16M4 12h16m-7 6h7"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
            <div
              class="hidden md:flex space-x-12 items-center"
            >
              <a
                href="/"
              >
                Home
              </a>
              <a
                href="/about"
              >
                About
              </a>
              <a
                href="/profile"
              >
                Profile
              </a>
               
              <div
                class="flex items-center space-x-4"
              >
                <img
                  alt="User Profile"
                  class="w-8 h-8 rounded-full"
                  src="https://example.com/avatar.png"
                />
                <button
                  class="border border-yellow-500 text-yellow-500 px-4 py-2 rounded-lg hover:bg-yellow-900 cursor-pointer"
                >
                  Logout
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="md:hidden overflow-hidden transition-all duration-300 ease-in-out max-h-0 opacity-0 bg-white dark:bg-black border-t border-gray-200 dark:border-gray-700"
        >
          <div
            class="px-4 py-3 flex flex-col gap-4"
          >
            <a
              href="/"
            >
              Home
            </a>
            <a
              href="/about"
            >
              About
            </a>
            <a
              href="/profile"
            >
              Profile
            </a>
             
            <div
              class="flex items-center space-x-4"
            >
              <img
                alt="User Profile"
                class="w-8 h-8 rounded-full"
                src="https://example.com/avatar.png"
              />
              <button
                class="border border-yellow-500 text-yellow-500 px-4 py-2 rounded-lg hover:bg-yellow-900 cursor-pointer"
              >
                Logout
              </button>
            </div>
          </div>
        </div>
      </nav>
    </header>
    <section
      class="absolute top-20 left-0 w-full text-center text-white px-6 z-30"
    >
      <h1
        class="text-4xl md:text-6xl font-bold"
        style="text-shadow: 0px 10px 15px rgba(0, 0, 0, 0.9);"
      >
        Meet Adventurers,
         
        <span
          class="text-yellow-400"
        >
          Explore Together
        </span>
      </h1>
      <p
        class="mt-4 text-lg md:text-xl text-gray-200"
        style="text-shadow: 0px 5px 10px rgba(0, 0, 0, 0.8);"
      >
        Find like-minded people to join you on your next adventure.
      </p>
    </section>
  </main>
</div>
`;
